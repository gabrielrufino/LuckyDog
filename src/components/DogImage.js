import React from 'react';
import {Image, StyleSheet, View} from 'react-native';

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#CACAAA',
    borderColor: '#C8AB83',
    borderRadius: 4,
    borderWidth: 4,
    flex: 1,
  },
  image: {
    borderRadius: 4,
    height: '100%',
  },
});

export default function DogImage({source}) {
  return (
    <View style={styles.container}>
      <Image style={styles.image} source={source} />
    </View>
  );
}
