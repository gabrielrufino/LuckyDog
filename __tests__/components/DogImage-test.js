import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import DogImage from '../../src/components/DogImage';

describe('components/DogImage.js', () => {
  it('Should render the DogImage correctly', () => {
    const source = {
      uri: 'https://images.dog.ceo//breeds//terrier-toy//n02087046_4455.jpg',
    };

    const tree = renderer.create(<DogImage source={source} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
