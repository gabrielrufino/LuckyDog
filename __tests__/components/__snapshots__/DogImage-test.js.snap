// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/DogImage.js Should render the DogImage correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#CACAAA",
      "borderColor": "#C8AB83",
      "borderRadius": 4,
      "borderWidth": 4,
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "https://images.dog.ceo//breeds//terrier-toy//n02087046_4455.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 4,
        "height": "100%",
      }
    }
  />
</View>
`;
